client/lobby/SelectionTab.h

```diff
@@ -95,6 +107,8 @@ class SelectionTab : public CIntObject
 	ESelectionScreen tabType;
 	Rect inputNameRect;
 
+	std::tuple<std::string, std::string, bool, bool> checkSubfolder(std::string path);
```

Comment:
please add comment, because signature is not clear: what returning tuple contains? Why path is not const reference?

Reply:
C++17 got std::string_view - should we start using that?

Reply:
Personally I would rather return struct here, with clearly named members instead of a tuple.

string_view - any actual need? IMO unless this is actually needed, e.g. for performance std::string is a better choice. string_view is non-owning, which *might* be faster, but also needs way more attention - since owning string might be destroyed or reallocated, invalidating any string_view's created from it

Reply:
when threading is not involved and string_view is used as function parameter then there should be no owning string destruction? string_view is easy to use and remember, if we think about new contributors

Reply:
and string is even easier to use since you don't have to correctly pick between string and string_view. Especially if we think about new contributors.

Reply:
Micro-optimization in vast majority of cases. Unless you're operating on strings several Mb in size.

At the very least this code is guaranteed to work. Even if it will take 1mks longer than code with const reference.

Reply:
For example, here we're discussing copying of 1 string, about 20-30 symbols long.

Meanwhile this code uses boost::split which requires construction of a vector and multiple strings inside said vector. As well as erasing front element from that vector, causing shift of all elements in it.

There is also at least one new redraw request which will force rendering of at least 800x600 pixels via CPU.

So effect of this oh so horrible string copy is zero. Sure, replacing it with const reference is not a bad idea, but nothing critically important for PR acceptance.

Reply:
I've changed it to struct. :)